From 277a7706ec07edc95073b7ab13a92ca4eddc56db Mon Sep 17 00:00:00 2001
From: Jamin Lin <jamin_lin@aspeedtech.com>
Date: Wed, 19 Jun 2024 15:02:03 +0800
Subject: [PATCH] Add header file path of hash-sigs package
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The hash-sigs opensource package is built by original Makeile.
Therefore, CMAKE cannot find hash-sigs package and get the
path of this package header file.

To get the path of this package header files, add “${YOCTO_STAGING_INCDIR_NATIVE}/hss”
in include_directories.

The YOCTO_STAGING_INCDIR_NATIVE variable is set by recipe and its value is
STAGING_INCDIR_NATIVE within Yocto build environment.

Signed-off-by: Jamin Lin <jamin_lin@aspeedtech.com>
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 14e87da..db24495 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,7 @@ endif ()
 add_compile_options(-g)
 include_directories(${OPENSSL_INCLUDE_DIR})
 include_directories(${LIBXML2_INCLUDE_DIRS})
+include_directories(${YOCTO_STAGING_INCDIR_NATIVE}/hss)
 add_executable(intel-pfr-signing-utility argparse.c cpld.c blocksign.c sslhelper.c main.c s_helpers.c log.c)
 add_executable(intel-pfr-signing-utility_lms argparse.c cpld.c blocksign.c sslhelper.c main.c s_helpers.c log.c)
 set_property(TARGET intel-pfr-signing-utility_lms PROPERTY COMPILE_DEFINITIONS LMS_SUPPORT)
-- 
2.25.1

